pf4j/src/main/java/org/pf4j/processor/ExtensionAnnotationProcessor.java

```diff
@@ -130,10 +130,10 @@ public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment
             }
         }
 
-        // write extensions
-        if (extensions.size() > 0) {
-            storage.write(extensions);
-        }
+        // even invoke if no extensions were found: an empty extensions descriptor
+        // is semantically correct and prevents classloaders falling back to descriptors
```

Comment:
> prevents classloaders falling back to descriptors from dependendencies

Can you add more details?

Reply:
I see now that the problem is described in https://github.com/pf4j/pf4j/issues/550.

Reply:
Indeed! I checked the patch in the context of my project and the extensions are looked up correctly again.